tests/plugins/test_pulp_pull.py

```diff
@@ -528,11 +530,121 @@ def get_manifest(request):
+        def get_manifest(request):
+            media_types = request.headers.get('Accept', '').split(',')
+            content_type = media_types[0]
+            return (200, {'Content-Type': content_type}, '{}')
+
+        url = re.compile(r'.*//crane.example.com/v2/.*/manifests/.*')
+        responses.add_callback(responses.GET, url, callback=get_manifest)
+        digests = {'digest': None}
+        workflow = self.workflow(
+            platform_descriptors=True,
+            postbuild_results={PLUGIN_GROUP_MANIFESTS_KEY: digests},
+            prebuild_results={PLUGIN_CHECK_AND_SET_PLATFORMS_KEY: set(platforms)},
+            expectv2schema2=expect_v2,
+        )
+        workflow.postbuild_plugins_conf = [{'name': 'pulp_sync'}]
+
+        not_found = requests.Response()
+        flexmock(not_found, status_code=requests.codes.not_found)
+        expectation = flexmock(requests.Session).should_receive('get')
+
+        # 1st retry don't find anything
+        for _ in range(5):
+            expectation = expectation.and_return(not_found)
+
+        expect_v1 = False
+        if 'x86_64' in platforms:
+            expect_v1 = True
+
+        # (for v1, v2, list.v2, oci, and oci.index media types) before get_manifest_digests
+        # v1
+        if expect_v1:
+            expectation.and_return(self.config_response_config_v1)
+        else:
+            expectation.and_return(not_found)
+
+        # v2
+        if found_v2:
+            expectation.and_return(self.config_response_config_v2)
+        elif expect_v1:
+            expectation.and_return(self.config_response_config_v1)
+        else:
+            expectation.and_return(not_found)
+
+        # v2_list
+        if found_v2_list:
+            expectation.and_return(self.config_response_config_v2_list)
+        else:
+            expectation.and_return(not_found)
+
+        # oci
+        if expect_v1:
+            expectation.and_return(self.config_response_config_v1)
+            expectation.and_return(self.config_response_config_v1)
+        else:
+            expectation.and_return(not_found)
+            expectation.and_return(not_found)
+
+        tasker = MockerTasker()
+        plugin = PulpPullPlugin(tasker, workflow, timeout=0.2,
+                                retry_delay=0.2, expect_v2schema2=expect_v2)
+        media_types = None
+
+        if manifest_list_only:
+            if found_v2_list:
+                media_types = plugin.run()
+            else:
+                with pytest.raises(Exception):
+                    plugin.run()
+                if expect_v1 or found_v2:
+                    assert "Expected schema 2 manifest list" in caplog.text()
+        else:
+            if found_v2_list and found_v2:
+                media_types = plugin.run()
+            else:
+                with pytest.raises(Exception):
+                    plugin.run()
+                if not found_v2_list and (found_v2 or expect_v1):
```

Comment:
The different conditionals in this test makes it very difficult to actually determine the expected behavior.